aperture.go

```diff
@@ -18,9 +18,12 @@ import (
 	gateway "github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
 	flags "github.com/jessevdk/go-flags"
 	"github.com/lightninglabs/aperture/auth"
+	"github.com/lightninglabs/aperture/challenger"
```

Comment:
commit message comment:

- `NewLndChallenger` isnt altered to take in an lndclient (which is the thing that would cause an import cycle), but rather is altered to take a `challenger.InvoiceClient` (which lnd client just happens to implement)
- nitty nit: check out LND contribution guidelines for ideal commit title & message structure 🙏

Reply:
also - for easier review, I think perhaps do 1 commit where you do the refactoring and then a separate commit for the `NewLndChallenger` change

Reply:
I moved the NewLndChallenger changes into a new commit. The change was a bit awkward because NewLndChallenger initially needs AuthConfig, which is defined in the main package. But then in the next commit we remove AuthConfig entirely because it's not needed once we update NewLndChallenger to take in the client as a parameter. This was the cleanest way I could think to make the change - maybe it's still worth it for review readability. Lmk your thoughts!